telegramServer.js

```diff
@@ -83,25 +84,44 @@ class TelegramServer extends EventEmitter {
   }
 
   async addUserMessage(message) {
+    await this.addUserUpdate(message, {message});
+    this.messageId++;
+    this.emit('AddedUserMessage');
+  }
+
+  async addUserCommand(message) {
+    assert.ok(message.entities, 'Command should have entities');
+    await this.addUserUpdate(message, {message, entities: message.entities});
+    this.messageId++;
+    this.emit('AddedUserCommand');
+  }
+
+  async addUserCallback(callbackQuery) {
+    await this.addUserUpdate(callbackQuery, {callbackQuery, callbackId: this.callbackId});
+    this.callbackId++;
+    this.emit('AddedUserCallbackQuery');
```

Comment:
BTW, I've changed `this.emit('AddedUserMessage');` → `this.emit('AddedUserCallbackQuery');` so it will comply with `this.emit('AddedUserCommand');`. But now I've notices that listener in `waitUserMessage` subscribed only to `waitUserMessage`. So currently it reacts on messages and callbacks but not the commands. But this `waitUserMessage` was never used in any tests or smth. So it is hard to tell what is it's original intent: should it react on any user input or only on messages?

Reply:
¯\\\_(ツ)_/¯ idunno, finally decided, that making it react on **any** user's input (4c035db5d5c4e55606f32fcf39556f107fa77ce9) is kinda less breaking change, and it is more likely that inconsistent behavior was introduced when commands and callbacks handling was added.